src/js/profile/chart.js

```diff
@@ -1,264 +1,486 @@
-    }
+  };
 
-    selectedGraphValueTypeChanged = (containerParent, index) => {
-        this.graphValueType = graphValueTypes[index];
-        this.triggerEvent('profile.chart.valueTypeChanged', this);
+  getChartTitle = (separator) => {
+    return this.selectedGroup === null ? `${this.title}` : `${this.title} by ${this.selectedGroup} ${separator} ${this.selectedFilter}`;
+  }
 
-        $(containerParent).find('.hover-menu__content_list a').each(function (itemIndex) {
-            $(this).removeClass('active');
-
-            if (index === itemIndex) {
-                $(this).addClass('active');
-            }
-        });
+  getPercentageValue = (currentValue, subindicators) => {
+    let percentage = 0;
+    let total = 0;
 
-        this.addChart();
+    for (const [label, value] of Object.entries(subindicators)) {
+      total += value.count;
     }
 
-    getPercentageValue = (currentValue, subindicators) => {
-        let percentage = 0;
-        let total = 0;
-
-        for (const [label, value] of Object.entries(subindicators)) {
-            total += value.count;
-        }
-
-        percentage = currentValue / total;
-
-        return percentage;
-    }
-
-    handleChartFilter = (indicators, groups, title) => {
-        let dropdowns = $(this.subCategoryNode).find('.filter__dropdown_wrap');
-        const filterArea = $(this.subCategoryNode).find('.profile-indicator__filters');
-
-        let siFilter = new SubindicatorFilter(indicators, filterArea, groups, title, this, dropdowns);
-        this.bubbleEvent(siFilter, 'point_tray.subindicator_filter.filter')
-
+    percentage = currentValue / total;
+
+    return percentage;
+  };
+
+  handleChartFilter = (indicators, groups, title) => {
+    let dropdowns = $(this.subCategoryNode).find(".filter__dropdown_wrap");
+    const filterArea = $(this.subCategoryNode).find(
+      ".profile-indicator__filters"
+    );
+
+    let siFilter = new SubindicatorFilter(filterArea, groups, title, this.applyFilter, dropdowns);
+    this.bubbleEvent(siFilter, "point_tray.subindicator_filter.filter");
+  };
+
+  applyFilter = (filteredData, selectedGroup, selectedFilter) => {
+    this.filteredData = filteredData;
+    this.selectedFilter = selectedFilter;
+    this.selectedGroup = selectedGroup;
+      this.setDownloadUrl(); //ToDo: use vega for getting data?
+      this.vegaView.signal('filterIndicator', selectedGroup)
+      this.vegaView.signal('filterValue', selectedFilter)
+      if(selectedGroup && selectedFilter !== "All values") {
+        this.vegaView.signal('applyFilter', true).run()
+      } else {
+        this.vegaView.signal('applyFilter', false).run()
+      }
+  };
+
+  getExportData = (isArray = false) => {
+    const data = this.vegaView.data('data_formatted');
+    const primaryGroup = this.vegaView.signal('mainGroup');
+    const formatting = this.vegaView.signal('numberFormat');
+    const datatypes  = this.vegaView.signal('datatype');
+    let rows = [];
```

Comment:
I think you could replace the whole function with `papaparse`

We don't need to and probably should not format the data anyway. 
we just need to add the raw data to the download.